sdk/go/src/sawtooth_sdk/client/encoder.go

```diff
@@ -0,0 +1,156 @@
+package client
+
+import (
+	"encoding/hex"
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	"sawtooth_sdk/protobuf/batch_pb2"
+	"sawtooth_sdk/protobuf/transaction_pb2"
+	"strings"
+	"time"
+)
+
+// See sdk/go/tests/client_test.go for example usage.
+
+type Transaction struct{ pb *transaction_pb2.Transaction }
+type Batch struct{ pb *batch_pb2.Batch }
+
+type TransactionEncoder struct {
+	privkey         []byte
+	pubkey          []byte
+	familyName      string
+	familyVersion   string
+	payloadEncoding string
+	addNonce        bool
+}
```

Comment:
I think you are right about not making assumptions about the defaults, so I am making changes based on that.

I don't think it makes sense to have a payloadEncoder option, at least not in go. Since Go is typed, this would require type casting at runtime by both the payloadEncoder and the NewTransaction functions. It also wouldn't benefit a developer that much, since they would be defining the function in either case.